src/pathbar_p.h

```diff
@@ -32,10 +34,13 @@ class PathButton: public QToolButton {
     QToolButton(parent),
     pathElement_(pathElement) {
 
-      setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+      setSizePolicy(QSizePolicy::Fixed, QSizePolicy::MinimumExpanding);
       setCheckable(true);
       setAutoExclusive(true);
       setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
+      /* respect the toolbar icon size (can be set with some styles) */
+      int icnSize = qApp->style()->pixelMetric(QStyle::PM_ToolBarIconSize);
```

Comment:
Just a note: by the use of `QApplication::style()` you're ignoring the `QWidget::style()` (if overriden...) and also, if style is changed (or some stylesheet used..) the toolbutton widget does not adapt.